app/helpers/cms_helper.rb

```diff
@@ -154,45 +154,45 @@ def cta_live_report
   end
 
   def get_resource_links 
-    resources = [
+    [
       get_resource(:resource_link_text, :resource_link_url, 'link', 'link-external'),
       get_resource(:resource_file_title, :resource_file, 'download', 'download')
     ].compact
   end
 
-  # Turns link[:url] value into a valid link if no http:// or https:// supplied
-  # Also sanitises it in the case of a download link
-  def linkify(url, fragment_link)
-    # If it is a file as opposed to a link
-    if fragment_link =~ /(file)/ 
-      file = @cms_page.fragments.find_by(tag: 'file')
-      return '' if !file || file.attachments.first.blank?
-      if Rails.env.development?
-        return rails_blob_path(file.attachments.first)
-      else
-        return file.attachments.first.service_url&.split('?')&.first
-      end
-    end
-    # TODO - add validations to Comfy pages controller to make it more robust 
-    # This is only temporary 
-    url =~ /(http:|https:)/ ? url : 'https://' + url
-  end
-
   private
 
   #  variation of resource link, if it is just a link, or whether it is a file
   # if those specific fragments for that page are present
   def get_resource(fragment_text, fragment_link, button_text, button_class)
-    if cms_fragment_content(fragment_text, @cms_page) && cms_fragment_content(fragment_link, @cms_page)
-      {
-        button: I18n.t("global.button.#{button_text}"),
-        classes: "button--#{button_class}",
-        text: cms_fragment_render(fragment_text, @cms_page),
-        title: cms_fragment_render(fragment_text, @cms_page),
-        url: linkify(cms_fragment_render(fragment_link, @cms_page), fragment_link)
-      }
+    frag_text, frag_link = @cms_page.fragments.where(identifier: [fragment_text, fragment_link])
+    return unless frag_text || frag_link
+
+    {
+      button: I18n.t("global.button.#{button_text}"),
+      classes: "button--#{button_class}",
+      text: cms_fragment_render(fragment_text, @cms_page),
+      title: cms_fragment_render(fragment_text, @cms_page),
+      url: linkify(frag_link)
+    }
+  end
+
+  # Turns link[:url] value into a valid link if no http:// or https:// supplied
+  # Also sanitises it in the case of a download link
+  def linkify(fragment_link)
+    if fragment_link.tag == 'file'
+      if Rails.env.development?
+        return rails_blob_path(fragment_link.attachments.first)
+      else
+        return fragment_link.attachments.first.service_url&.split('?')&.first
+      end
     else
-      nil
+      begin
+        url = URI.parse(fragment_link.content)
+        (url.is_a?(URI::HTTP) && !url.host.nil?) ? url.to_s : 'Invalid or missing URL'
```

Comment:
It's also appending these messages to the root URL as part of the generated URL - will need to have a think.

Reply:
So the intent here was to return something that is not a URL (so a normal string) so that it won't create, possibly, any link. Perhaps what you are mentioned is something to be done on the frontend and ideally change those strings into a default value that could be more easily parsed by the frontend itself?

Reply:
I've made a couple of changes to the helper and put in a condition in the partial to only display the list item with the link within if the URL is not blank.

Reply:
If this approach is sound, I can merge this in.